datafusion/substrait/src/logical_plan/producer.rs

```diff
@@ -1194,7 +1194,24 @@ pub fn to_substrait_rex(
                             col_ref_offset,
                             extensions,
                         )?)),
-                        failure_behavior: 0, // FAILURE_BEHAVIOR_UNSPECIFIED
+                        failure_behavior: 2, // FAILURE_BEHAVIOR_THROW_EXCEPTION
+                    },
+                ))),
+            })
+        }
+        Expr::TryCast(TryCast { expr, data_type }) => {
+            Ok(Expression {
+                rex_type: Some(RexType::Cast(Box::new(
+                    substrait::proto::expression::Cast {
+                        r#type: Some(to_substrait_type(data_type, true)?),
+                        input: Some(Box::new(to_substrait_rex(
+                            state,
+                            expr,
+                            schema,
+                            col_ref_offset,
+                            extensions,
+                        )?)),
+                        failure_behavior: 1, // FAILURE_BEHAVIOR_RETURN_NULL
```

Comment:
minor: This predates your PR, but it think it would be good to use something like
```rust
FailureBehavior::ReturnNull.into()
```
here, which allows us to set the value of `failure_behaviour` directly from the generated enum value.